src/ConstructiveSolidGeometry/SurfacePrimitives/RegularPolygon.jl

```diff
@@ -62,6 +62,33 @@ end
 get_r_limits(rp::RegularPolygon) = (_left_radial_interval(rp.r), _right_radial_interval(rp.r))
 get_r_at_φ(rp::RegularPolygon{N,T}, φ::T) where {N,T} = get_r_limits(rp) .* T(cos(π/N)/cos(π/N - mod(φ, 2π/N)))
 
+function get_y_at_x(r::T, corners, g::CartesianTicksTuple{T}, x::T)::T where {T}
+    for i in eachindex(corners)[1:end-1]
+        s1,c1 = r .* corners[i]
+        s2,c2 = r .* corners[i+1]
+        if isapprox(c1, c2, atol = geom_atol_zero(T))
+            return isapprox(x, c1, atol = geom_atol_zero(T)) ? max(s1,s2) : T(0)
+        end
+        t = (x-c1)/(c2-c1)
+        if t in 0..1 return s1 + t * (s2-s1) end
+    end
+    ()
+end
+
+function get_missing_x_at_y(r::T, corners::SVector{N, Tuple{T,T}}, g::CartesianTicksTuple{T}, y::T) where {N,T}
+    x = []
+    for i in eachindex(corners)[1:end-1]
+        s1,c1 = r .* corners[i]
+        s2,c2 = r .* corners[i+1]
+        if ! isapprox(s1, s2, atol = geom_atol_zero(T))
+            t = (y-s1)/(s2-s1)
+            if t in 0..1 push!(x, c1 + t * (c2-c1)) end
+        end
+    end
+    x
+end
```

Comment:
Is there an elegant way to write this in Array syntax like we do it for sampling primitives?